mathml/presentation-markup/too-few-children.html

```diff
@@ -0,0 +1,455 @@
+<script src="/mathml/support/feature-detection.js"></script>
+<script src="/mathml/support/layout-comparison.js"></script>
+<script>
+  function getBox(id, i) {
+    return document.getElementById(id+i)
+  }
+
+  function removeIgnoredMrows(children) {
+    let result = [];
+    Array.from(children).forEach(child => {
+      if (child.getAttribute("class")==="ignore_mrow") {
+        if (child.childElementCount > 0)
+          Array.prototype.push.apply(result, child.children);
+      } else {
+        result.push(child);
+      }
+    });
+    return result;
+  }
+
+
+  function compareLayout(element, reference, epsilon) {
+    // Compare sizes of elements and children.
+    compareSize(element, reference, epsilon);
+    const elementBox = element.getBoundingClientRect();
+    const referenceBox = reference.getBoundingClientRect();
+
+    const elementChildren = childrenParticipatingToLayout(element);
+    const referenceChildren = removeIgnoredMrows(childrenParticipatingToLayout(reference));
+    
+    if (elementChildren.length != referenceChildren.length) {
+      throw "Reference should have the same number of children participating to layout."
+    }
+
+
+    for (let i = 0; i < elementChildren.length; i++) {
+        const childBox = elementChildren[i].getBoundingClientRect();
+        const referenceChildBox = referenceChildren[i].getBoundingClientRect();
+        compareLayout(childBox, referenceChildBox, epsilon);
+    }
+  }
+
+
+  setup({ explicit_done: true });
+  window.addEventListener("load", () => { document.fonts.ready.then(runTests); });
+
+  function runTests() {
+
+    // elements being tested
+    const elements = [
+      {name:"mover", n:2},
+      {name:"munder", n:2},
+      {name:"munderover", n:3},
+      {name:"msub", n:2},
+      {name:"msup", n:2},
+      {name:"msubsup", n:3},
+      {name:"mmultiscripts", n:1},
+      {name:"mfrac", n:2},
+      {name:"mroot", n:2},
+      {name:"msqrt", n:1},
+      {name:"mstyle", n:1},
+      {name:"merror", n:1},
+      {name:"mpadded", n:1},
+      {name:"mphantom", n:1},
+      {name:"menclose", n:1},
+      {name:"mtable", n:1},
+      {name:"mtr", n:1},
+      {name:"mtd", n:1},
+      {name:"maction", n:1},
+      {name:"semantics", n:1},
+      {name:"math", n:1}
+    ];
+
+    // add a has_xxx key/value pair for feature dectection
+    elements.forEach((e) => { e.has = "has_"+e.name;});
+
+    // preconditions for all the tests running
+    // note: these could be done for each test, but repeatitive and really this is independent of what is really tested
+    test(
+      function() {
+        assert_true(MathMLFeatureDetection.has_mspace());
+        assert_true(MathMLFeatureDetection.has_mrow());
+        // check all tested elements (not all has_xxx functions implemented yet)
+        // elements.forEach(e) (
+        //  assert_true(MathMLFeatureDetection[e.has])
+        // )
+      },
+      "features/preconditions for running empty MathML tests"
+    );
```

Comment:
This should be removed. The test() are really independent from each other, so this will make this one test fail for browsers lacking mathml layout but does not affect at all the result in the other tests.